charts/dev/livebook/values.yaml

```
@@ -19,12 +19,30 @@ service:
     ports:
       main:
         port: 8787
-        protocol: tcp
+        targetPort: 8080
+  headless:
+    enabled: true
+    primary: false
+    clusterIP: None
+    ports:
+      headless:
+        enabled: true
+        port: 8787
         targetPort: 8080
 workload:
   main:
     podSpec:
       containers:
         main:
           env:
-            LIVEBOOK_PASSWORD: livebookiscool
+            # A_ because these variables are sorted alphabetically and this one needs to come first.
+            A_POD_IP:
+              fieldRef:
+                fieldPath: status.podIP
+            LIVEBOOK_CLUSTER: '{{ printf "dns:%s-headless.%s.svc.cluster.local" (include "tc.v1.common.lib.chart.names.fullname" .) .Release.Namespace }}'
+            LIVEBOOK_DISTRIBUTION: name
+            LIVEBOOK_NODE: livebook@$(A_POD_IP)
+            LIVEBOOK_PORT: "{{ .Values.service.main.ports.main.targetPort }}"
```

```suggestion
            LIVEBOOK_PORT: "{{ .Values.service.main.ports.main.port }}"
```

set it like this and remove the `targetPort` fromt he service